pydantic/_internal/_fields.py

```diff
@@ -110,6 +111,8 @@ def collect_model_fields(  # noqa: C901
             continue
         if ann_name.startswith('_'):
             continue
+        if cls.__pydantic_root_model__ and ann_name != 'root':
+            continue
```

Comment:
I think we should raise an error here

Reply:
A test case that hits this line would definitely help. Also, I agree that the use case doesn't make sense. One can use a private attribute if desired.

Reply:
Test case:
```python
class MyModel(RootModel[int]):
    x: str
```
does this hit it? I'll try, but that's what I was thinking would

Reply:
Yep I confirmed it does:
```
from pydantic import RootModel


class MyModel(RootModel[int]):
    x: str

print(MyModel.model_fields)
#> {'root': FieldInfo(annotation=int, required=True)}
```
and adding `assert False` on this line makes the above error.

I would suggest the following:
```suggestion
            raise TypeError('Cannot add additional fields to subclasses of RootModel`)
```

Reply:
Done